### PR TITLE
Bill "subItems" schema

### PR DESCRIPTION
**Pull request by denwilliams**
Not particularly happy with this.

Putting it up for review.

### CODE REVIEW

File: itemv2.js [hunk 1]
```diff
@@ -70,7 +89,7 @@ const BaseItem = {
       type: "integer",
       title: "The cost of a single instance of this item.",
       description:
-        "The cost is represented in the lowest denomination (eg cents). For discounts this should be negative.",
+        "The cost is represented in the lowest denomination (eg cents). For discounts this should be negative. Excludes the cost of options.",
```
Comment (denwilliams) [inline itemv2.js:73]: Feel it makes more sense for amount to work the same way as order items - to represent the base price, excluding options.

File: itemv2.js [hunk 2]
```diff
@@ -100,7 +120,12 @@ const BaseItem = {
 
 const BillItem = {
   type: "object",
-  properties: Object.assign({}, BaseItem.properties, {}),
+  properties: Object.assign({}, BaseItem.properties, {
+    changes: {
```
Comment (denwilliams) [inline itemv2.js]: Didn't use the word options, as it is a different schema to order options. Feel that is confusing.

Can't use `meta` as it is more than just meta, you need it to calculate the entire price.

Opted for `changes`.

File: itemv2.js (hunk 1 above)
Comment (smithamax) [reply itemv2.js:73]: Yeah okay that works for me.

File: itemv2.js (hunk 2 above)
Comment (smithamax) [reply itemv2.js]: Yeah changes isn't bad, better than `additions` which was my first though. The only other thing I can think of is `subItems`, probably more representative, but kind of a crappy name.

File: itemv2.js [hunk 3]
```diff
@@ -100,7 +120,12 @@ const BaseItem = {
 
 const BillItem = {
```
Comment (smithamax) [inline itemv2.js]: Can we move this and `SimpleOption` to the bill file, keeping it all together. We can just have the base Item here, if we even still need it, they've diverged enough now that maybe they don't need to inherit the same props. That being said, consistent props like `name` and `amount` are nice.

File: itemv2.js (hunk 2 above)
Comment (denwilliams) [reply itemv2.js]: I don't mind `subItems` actually

**Review by smithamax — APPROVED** [review]

**Pull request merged**